text/0010-exception-types.md

```diff
@@ -0,0 +1,119 @@
+* Start Date: 2022-09-17
+* RFC Type: feature
+* RFC PR: 0010
+* RFC Status: draft
+
+# Summary
+
+This RFC suggests a feature which introduces additional types of exceptions next to `mechanism.handled`.
+
+# Motivation
+
+Currently, exception which cause the running software to exit (the process died/hard crash) are marked as `handled: false`. This isn't enough for SDKs where an exception can be unhandled but at the same time doesn't cause the software to exit.
+
+Sentry shows exceptions which aren't caught by the developer (unhandled) but also do not cause the software to exit in the same way as exceptions which are manually caught by the developer. This seems rather unintuitive and makes exceptions seem less severe than they are.
```

Comment:
this seem inaccurate to me. Or maybe I'm not understanding.

Sentry doesn't show exceptions that are not captured by the user (unhandled) in the same way it shows exceptions  exceptions manually captured (handled). One has the red banner:
<img width="108" alt="image" src="https://user-images.githubusercontent.com/1633368/192897344-49dcdbfe-d67c-49cd-ace3-0b11b40652c8.png">

Do you mean that Sentry shows all unhandled errors in the same way, independent if they caused the app to crash?

Reply:
Unhandled exceptions in Flutter don't have that red banner because they don't crash the application. I'm not sure how similar behavior is implemented in other SDKs.

That means there's no visual differentiation between handled and unhandled exceptions as long as they don't cause the software to exit.

> Do you mean that Sentry shows all unhandled errors in the same way, independent if they caused the app to crash?

No, unhandled exceptions which cause the software to exit are shown visually different.

Reply:
Right now, Handled and Unhandled exceptions are `handled: null` or `handled: true` so there's no red banner, no way to filter or alert them differently.
Only when it crashes on the Native side (C code or Native threads), the flag is `handled: false`.
So yes, it's bad and the RFC tries to fix this.